### ISSUE TITLE
change the interpretation of `rel & {cond1, cond2}` from `AndList` to `OrList`

### ISSUE THREAD

**Issue by dimitri-yatsenko** [datajoint/datajoint-matlab#96]
Until now, restricting by a list (or a set) resulted in the application of all the restrictions conjunctively   , i.e. `rel & {cond1, cond2}` was equivalent to `rel & cond1 & cond2` or `rel & dj.AndList(cond1, cond2)`.

I propose to change this interpretation to adjunctive interpretation so that the above expression will be equivalent to `rel & dj.OrList(cond1, cond2)`.

This change makes restrictions by lists logically consistent with the semijoin `rel & otherRel` since `otherRel` may be thought of as a set of conditions.  The semijoin is applied adjunctively.

**Comment by dimitri-yatsenko** [comment]
This may cause problems in existing code base if anyone has restricted by cell arrays intending all the conditions to be met.

**Comment by eywalker** [comment]
I agree with this change. Indeed this may break codes if it already depended on this behavior, but all in all cell list acting as `or-list` is more consistent with how restriction with structure array works.

**Comment by ixcat** [comment]
should be implemented with #77

**Comment by shenshan** [comment]
In our documentation, it is explained as an OR List

https://docs.datajoint.io/matlab/queries/06-Restriction.html#restriction-by-a-table

**Comment by ixcat** [comment]
> In our documentation, it is explained as an OR List
> 
> https://docs.datajoint.io/matlab/queries/06-Restriction.html#restriction-by-a-table

Hi @shenshan - this is noted in the block: ```This section documents future intended behavior in MATLAB, which is contrary to current behavior. DataJoint for MATLAB has an open issue tracking this change.``` which links to https://github.com/datajoint/datajoint-matlab/issues/128 , marked as duplicate of this issue.

**Comment by SchwartzNU** [comment]
Is there a fix coming soon or a workaround for this? How can one make an OR list in MATLAB?

**Comment by ixcat** [comment]
Hello -
For the moment, there are 2 options:
1) A structured array is interpreted as an OrList
2) String based restriction using 'or' clauses e.g. `some_id=1 OR some_id=2`

**Comment by SchwartzNU** [comment]
Thanks. That is very helpful. I didn't realize the 'OR' clauses worked like that. That will allow me to build OR queries with operators like '>' that are not supported in a struct.
